### PR TITLE
Add body tracking sample

### PR DESCRIPTION
**Pull request by cabanier**
FYI @toji
example video:
https://github.com/immersive-web/webxr-samples/assets/1513308/6e5227c6-d451-4846-a0a2-6607bfcce627

### CODE REVIEW

**Comment by Maksims** [comment]
This is amazing!
I had to implement upper body IK multiple times, and it is not an easy task, and falls short when estimating shoulder twist.
Also legs! Yes please!

File: proposals/body-tracking.html [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+            let i = 0;
+            const matrix = new THREE.Matrix4();
+            let space = renderer.xr.getReferenceSpace();
+            body.forEach(part => {
+
+              const pose = frame.getPose(part, space);
+              const position = pose.transform.position;
+
+              if (!part.jointName.includes("hand")) {
+                matrix.copy(scalehand);
```
Comment (toji) [inline proposals/body-tracking.html:162]: Why do hands need to be scaled up 3x? In your video it looks like the opposite, and the hands are scaled down?

Comment (toji) [reply proposals/body-tracking.html:162]: Oh, wait, I see. This is checking to ensure that "hand" *isn't* part of the name. That's pretty confusing. Can you invert the logic here?

File: proposals/body-tracking.html [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+              spheres.instanceMatrix.needsUpdate = true;
+
+            }
+
+            let i = 0;
+            const matrix = new THREE.Matrix4();
+            let space = renderer.xr.getReferenceSpace();
+            body.forEach(part => {
+
+              const pose = frame.getPose(part, space);
```
Comment (toji) [inline proposals/body-tracking.html:158]: Do we want to encourage people to use `fillPoses()` here? Or does that introduce too many complications for an otherwise simple sample.

**Review by toji — COMMENTED** [review]
As you said, this is just an FYI for now but I've got some comments for you anyway!

File: proposals/body-tracking.html [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+          onEndSession: onEndSession,
+          textEnterXRTitle: "START AR",
+          textXRNotFoundTitle: "AR NOT FOUND",
+          textExitXRTitle: "EXIT AR",
+        });
+
+        document.querySelector('header').appendChild(xrButton.domElement);
+
+        if (navigator.xr) {
+          navigator.xr.isSessionSupported('immersive-ar')
```
[the remark this answers is not in the document]
Comment (yuripourre) [reply proposals/body-tracking.html:75]: @toji I am very curious about this too.

@cabanier Are you using some sort of polyfill? Or the API is giving you that information?

File: proposals/index.html [hunk 4]
```diff
@@ -132,6 +132,9 @@ <h2 class='tagline'>Proposals</h2>
               description: 'Demonstrates use of the mesh detection API in an immersive-ar session. ' +
                            'Implements JavaScript-level hit-test on the meshes and leverages the Anchors API.' },
 
+            { title: 'Body tracking', category: 'AR',
```
[the remark this answers is not in the document]
Comment (cabanier) [reply proposals/index.html:135]: Correct, just like hands this will also work in VR. I'll code something up

File: proposals/body-tracking.html (hunk 2 above)
Comment (cabanier) [reply proposals/body-tracking.html:158]: yeah. I forgot about that call :-)
I will update the code

File: proposals/body-tracking.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (yuripourre) [reply proposals/body-tracking.html:75]: @cabanier Sorry, let me rephrase. Is the WebXR implementation in the browser providing you all the information for the body tracking? Using the `body-tracking` feature? Or you are generating this information somewhere else and injecting in the browser? I am asking because `body-tracking` is not officially described/supported by the WebXR specification.

I am just very curious to understand where the bones positions/rotations comes from.

[the remark this answers is not in the document]
Comment (cabanier) [reply proposals/body-tracking.html:75]: > @cabanier Sorry, let me rephrase. Is the WebXR implementation in the browser providing you all the information for the body tracking? Using the `body-tracking` feature? Or you are generating this information somewhere else and injecting in the browser?

It's a new extension to WebXR that I'm proposing: https://cabanier.github.io/webxr-body-tracking/
It uses Meta's body tracking OpenXR implementation under the hood.

[the remark this answers is not in the document]
Comment (yuripourre) [reply proposals/body-tracking.html:75]: @cabanier Thanks for explaining. Nice work, I hope it's accepted! 🤞

[the remark this answers is not in the document]
Comment (cabanier) [reply proposals/body-tracking.html:116]: > Is this functionality only available in AR on Quest? If not, is there any way can we make this demo work in either AR or VR?

I can make it work in vr as well. Maybe in a follow up?

File: proposals/body-tracking.html [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+          const scaledMatrix = new THREE.Matrix4();
+          let space = renderer.xr.getReferenceSpace();
+          frame.fillPoses(body.values(), space, positions);
+          body.forEach(part => {
+            matrix.fromArray(positions.slice(i * 16, (i + 1) * 16));
+
+            if (!part.jointName.includes("hand")) {
+              scaledMatrix.makeScale(3, 3, 3);
+            } else {
+              -             scaledMatrix.identity();
```
Comment (toji) [inline proposals/body-tracking.html:232]: What's the `-` here? Typo or leftover from a merge? Does this code run?
